### PR TITLE
[Event Grid Client] Overview README

### PR DESCRIPTION
**Pull request by jsquire**
# Summary

The focus of these changes is to add the overview content as a README in the root of the SDK directory, which will be used by the MS Docs site to as the landing page: [Azure Event Grid libraries for .NET - Azure for .NET Developers](https://docs.microsoft.com/dotnet/api/overview/azure/eventgrid).

# Related and References

- [Overview page for Event Grid ref docs show older packages (#21125)](https://github.com/Azure/azure-sdk-for-net/issues/21125)
- [Event Hubs Overview README](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventhub/README.md)
- [Key Vault Overview (no README)](https://docs.microsoft.com/dotnet/api/overview/azure/key-vault?view=azure-dotnet)

### CODE REVIEW

File: sdk/eventgrid/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# Azure Event Grid libraries for .NET
+
+Azure Event Grid allows you to build applications with event-based architectures. The Event Grid service fully manages all routing of events from any source, to any destination, for any application. Azure service events and custom events can be published directly to the service, where the events can then be filtered and sent to various recipients, such as built-in handlers or custom webhooks. To learn more about Azure Event Grid: [What is Event Grid?](https://docs.microsoft.com/azure/event-grid/overview)
+
+## Libraries for data access
+
+The current generation of the Azure Event Grid client library uses versions 4.0 and above.  Microsoft recommends using version 4.2 or higher for new applications.  If you are unable to existing applications to version 4.x, then Microsoft recommends using version 3.2 or higher.
```
Comment (JoshLove-msft) [inline sdk/eventgrid/README.md:7]: Is it worth clarifying here that we are talking about different packages?

Comment (JoshLove-msft) [inline sdk/eventgrid/README.md:7]: Probably should avoid mentioning 4.2 as this will quickly become out of date and will be a pain to maintain.

**Review by JoshLove-msft — APPROVED** [review]

File: sdk/eventgrid/README.md (hunk 1 above)
Comment (jsquire) [reply sdk/eventgrid/README.md:7]: Yeah... it is.   I copy/pasted this from Event Hubs, but I don't remember why it was done this way.   I think I'm going to fix both places.

**Pull request merged**